merkle/src/store.rs

```diff
@@ -1,33 +1,71 @@
 use failure::Error;
-use merkle::Element;
+use merkle::{Element, next_pow2};
 use positioned_io::{ReadAt, WriteAt};
-use std::fs::File;
+use std::fs::{File, OpenOptions};
+use std::path::{Path, PathBuf};
 use std::marker::PhantomData;
 use std::ops::{self, Index};
 use tempfile::tempfile;
 
 pub type Result<T> = std::result::Result<T, Error>;
 
+
+#[derive(Clone, Debug)]
+pub struct StoreConfig {
+    /// A directory in which data (a merkle tree) can be persisted.
+    pub path: String,
+
+    /// A unique identifier used to help specify the on-disk store
+    /// location for this particular data.
+    pub id: String,
+
+    /// The number of merkle tree levels above the base to cache on disk.
+    pub levels: usize,
+}
+
+impl StoreConfig {
+    pub fn new(path: String, id: String, levels: usize) -> StoreConfig {
+        StoreConfig {
+            path,
+            id,
+            levels
+        }
+    }
+
+    // Deterministically create the data_path on-disk location from a
+    // path and specified id.
+    pub fn data_path(path: &str, id: &str) -> PathBuf {
+        Path::new(&path).join(format!("sc-data-{}.dat", id))
```

Comment:
lets do sth like `"sc-01-data-{}.dat"`, that way we have a version thing in there